src/react-web3/Web3Provider.js

```diff
@@ -12,72 +12,95 @@ class Web3Provider extends Component {
+      return <Loader show={true} />
     }
-    return ( renderContainer )
 
+    if (this.state.render && (!this.state.web3 || !this.state.approvePermissions)) {
+      return <Web3UnavailableScreen />
+    }
+
+    return null
   }
 
   componentDidMount() {
     this.checkWeb3()
-    this.initPoll()
-
-    // Wait 1 second to render
-    setTimeout(function() {
-      this.setState({render: true})
-    }.bind(this), 1500)
+      .then(this.initPoll())
+      .then(() => {
+        this.setState({
+          render: true
+        })
+      })
+      .catch(err => {
+        console.log(err)
+        this.setState({
+          render: true
+        })
+      })
   }
 
   checkWeb3() {
     const setWeb3 = () => {
-      window.web3 = new Web3(window.web3.currentProvider)
-      this.setState({
-        web3: window.web3
-      })
-      this.fetchAccounts()
+      try {
+        window.web3 = new Web3(window.web3.currentProvider)
+        this.setState({
+          approvePermissions: true,
+          web3: window.web3
+        })
+      } catch (err) {
+        console.log('There was a problem fetching accounts', err)
+      }
     }
 
-    window.addEventListener('load', () => {
-      const { ethereum } = window
-      // Modern dapp browsers...
-      if (ethereum) {
-        window.web3 = new Web3(ethereum)
-        try {
-          // Request account access if needed
-          ethereum.enable().then(()=> {
-            if (!this.state.web3) {
-              setWeb3()
-            }
-          })
-        } catch (error) {
-          // User denied account access...
-          console.log('User denied account access')
+    return new Promise((resolve, reject) => {
+      try {
+        const { ethereum } = window
+        // Modern dapp browsers...
+        if (ethereum) {
+          window.web3 = new Web3(ethereum)
+          try {
+            // Request account access if needed
+            ethereum.enable().then(() => {
+              if (!this.state.web3) {
+                setWeb3()
+              }
+            })
+            resolve()
+          } catch (error) {
+            // User denied account access...
+            const msj = 'User denied account access'
+            throw new Error(msj)
+          }
         }
-      }
-      // Legacy dapp browsers...
-      else if (window.web3) {
-        setWeb3()
-      }
-      // Non-dapp browsers...
-      else {
-        console.log('Non-Ethereum browser detected. You should consider trying a wallet!')
+        // Legacy dapp browsers...
+        else if (window.web3) {
+          setWeb3()
+          resolve()
+        }
+        // Non-dapp browsers...
+        else {
+          const msj = 'Non-Ethereum browser detected. You should consider trying a wallet!'
+          throw new Error(msj)
+        }
+      } catch (err) {
+        this.setState({
+          approvePermissions: false
+        })
+        console.log(err.message)
```

Comment:
`logger`

Reply:
@fernandomg 
If my memory doesn't fail me, `logger` is used in the branch 2.0, please correct me if I'm wrong

Greetings